presto-jdbc/src/main/java/com/facebook/presto/jdbc/PrestoConnection.java

```diff
@@ -741,6 +770,31 @@ WarningsManager getWarningsManager()
         return warningsManager;
     }
 
+    PrestoResultSet invokeQueryInterceptorsPre(String sql, Statement interceptedStatement)
+    {
+        return processInterceptors(interceptor -> interceptor.preProcess(sql, interceptedStatement), null);
+    }
+
+    PrestoResultSet invokeQueryInterceptorsPost(String sql, Statement interceptedStatement, PrestoResultSet originalResultSet)
+    {
+        return processInterceptors(interceptor -> interceptor.postProcess(sql, interceptedStatement, originalResultSet), originalResultSet);
+    }
```

Comment:
in this function - we are passing a fixed reference to the originalResultSet rather than the one returned by the previous post processor - in my opinion we should ensure that the correct and latest result set is being passed